### PR TITLE
Minor: Updated dockerfile

### LINKED ISSUE #16: Implement Dockerfile to build Alpine Linux

### PR DESCRIPTION
**Pull request by Oleksandr6676**
added new instructions to Dockerfile
Closes #16

### CODE REVIEW

**Comment by Oleksandr6676** [comment]
Hi Jan. I've added some instructions to dockerfile. So now this dockerfile builds an image based on latest alpine with pre-installed python 3 and pip. Anything else I could possibly add here?

**Comment by Oleksandr6676** [comment]
Hi Jan. I've added some instructions to dockerfile. So now this dockerfile builds an image based on latest alpine with pre-installed python 3 and pip. Anything else I could possibly add here?

File: Dockerfile [hunk 1]
```diff
@@ -1 +1,12 @@
+FROM alpine:latest
+
+LABEL description="Alpine Linux with Python 3"
+
```
Comment (JanMate) [inline Dockerfile:4]: Add env var PYTHONPATH. It should contain python binary and path to python module in this repo.

File: Dockerfile [hunk 2]
```diff
@@ -1 +1,12 @@
...
+WORKDIR /usr/src/app
+
+RUN apk upgrade && \
+	apk add python3 && \
+	apk add py3-pip
+
```
Comment (JanMate) [inline Dockerfile:10]: Set a new 'core' user to avoid using root privileges due to security reasons.

File: Dockerfile [hunk 3]
```diff
@@ -1 +1,12 @@
+FROM alpine:latest
+
+LABEL description="Alpine Linux with Python 3"
+
+WORKDIR /usr/src/app
```
Comment (JanMate) [inline Dockerfile:5]: Before working dir copy python module with code into the image.

**Review by JanMate — CHANGES_REQUESTED** [review]
Nice start! Let's improve it a little.

File: Dockerfile [hunk 4]
```diff
@@ -1 +1,12 @@
+FROM alpine:latest
```
[the remark this answers is not in the document]
Comment (Oleksandr6676) [reply Dockerfile:1]: Yes, sure. I will use it

File: Dockerfile [hunk 5]
```diff
@@ -1 +1,12 @@
...
+
+LABEL description="Alpine Linux with Python 3"
+
+WORKDIR /usr/src/app
+
+RUN apk upgrade && \
+	apk add python3 && \
+	apk add py3-pip
+
+CMD python3 --version
```
[the remark this answers is not in the document]
Comment (Oleksandr6676) [reply Dockerfile:11]: I don't know to be honest, but I'll try to figure it out :)

File: Dockerfile (hunk 1 above)
Comment (Oleksandr6676) [reply Dockerfile:4]: Do you mean the path to the copied python module inside the image, right?
Something like this ENV PYTHONPATH="$PYTHONPATH:/path-to-the-module"

Comment (JanMate) [reply Dockerfile:4]: Exactly, but there can be only path to python binary and the module path.

File: Dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (JanMate) [reply Dockerfile:11]: Check this out: https://docs.docker.com/engine/reference/builder/#entrypoint

File: Dockerfile [hunk 6]
```diff
@@ -1 +1,19 @@
...
+
+RUN adduser -S -D New_user
+USER New_user
+
+ENV PYTHONPATH="/usr/local/bin/python;/app/core"
+
+COPY requirements.txt /app/requirements.txt
+WORKDIR app
+RUN pip install -r requirements.txt
+COPY core core
```
Comment (JanMate) [inline Dockerfile:16]: Please, keep copy commands together.

**Review by JanMate — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 7]
```diff
@@ -1 +1,19 @@
+FROM python:3.8-alpine
+
+LABEL description="Alpine Linux with Python 3" \
+	  author="Oleksandr6676" \
+	  app="Core app" \
+      version=1.0.0
+
+RUN adduser -S -D core
```
Comment (JanMate) [inline Dockerfile:8]: Do you know why our user must be a system one?

File: Dockerfile [hunk 8]
```diff
@@ -1 +1,19 @@
+FROM python:3.8-alpine
+
+LABEL description="Alpine Linux with Python 3" \
+	  author="Oleksandr6676" \
+	  app="Core app" \
```
Comment (JanMate) [inline Dockerfile:5]: Please, fix indentation. Then we can merge.

**Review by JanMate — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 8 above)
Comment (Oleksandr6676) [reply Dockerfile:5]: Hmm, it looks fine in my code editor. Sure, I’ll fix it

Comment (JanMate) [reply Dockerfile:5]: Interesting, you are right. So let it be then.

**Comment by JanMate** [comment]
@Oleksandr6676 will you work on this PR additionally? Or can I merge it?

**Comment by Oleksandr6676** [comment]
> @Oleksandr6676, will you work on this PR additionally? Or can I merge it?

Hi Jan! Yes you can merge it, thank you!

**Review by JanMate — APPROVED** [review]

**Pull request merged**